packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -73,30 +73,48 @@ class ErrorHandlingFileSystem extends ForwardingFileSystem {
   /// This method should be preferred to checking if it exists and
   /// then deleting, because it handles the edge case where the file or directory
   /// is deleted by a different program between the two calls.
-  static bool deleteIfExists(FileSystemEntity file, {bool recursive = false}) {
+  static bool deleteIfExists(FileSystemEntity file, {bool recursive = false, bool handleSharingViolation = false}) {
```

Comment:
How about something like this? I'm not sure if the `_noExitOnFailure` case should be skipped when `continueOnError` is true, though. I'm not quite sure how that's supposed to work.

```dart
static bool deleteIfExists(
  FileSystemEntity file, {
  bool recursive = false,
  bool continueOnError = true,
  String errorMessage = "some default",
}) {
  if (!file.existsSync()) {
    return false;
  }
  try {
    file.deleteSync(recursive: true);
  } on FileSystemException catch (err) {
    if (_noExitOnFailure) {
      rethrow;
    }

    final int errorCode = err?.osError?.errorCode;
    const int kSystemCannotFindFile = 2;

    // Detect the case where success is reported as failure.
    if (errorCode == kSystemCannotFindFile) {
      ...
    }

    // For everything else, respect `continueOnError`
    if (continueOnError) {
      printError(...errorMessage...)
    } else {
      throwToolExit(...errorMessage...)
    }
  }

  return true;
}
```

Reply:
I'm not really sure how continueOnError/message fits into the existing use cases - I think this method is getting a bit too big and I think a larger change isn't really cherry pickable.

Reply:
You're right about the size for the cherry-pick.

`continueOnError` doesn't really try to fit in with existing use cases. The idea is to only crash the tool if we really, really care that a file isn't getting deleted, which would be new, instead of handling all these failures on a case-by-case basis.